DevicesDistributor.Core/Services/Concrete/DeviceRepository.cs

```diff
@@ -0,0 +1,29 @@
+﻿using DevicesDistributor.Core.Entities;
+using DevicesDistributor.Core.Services.Abstract;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DevicesDistributor.Core.Services.Concrete
+{
+    public class DeviceRepository : IDeviceRepository
+    {
+        private  readonly IDictionary<Guid, Device> _entities = new Dictionary<Guid, Device>();
+
```

Comment:
unikaj nadmiarowych enterów. Odzielanie elementów pojedyńczym enterem jest ok, ale co za duzo to nie zdrowo

Ale nie to jest największym problemem. Masz zdublikowane dane, bo na jedną encję są 2 repozytoria. Są sytuacje w których tak trzeba, ale to nie ta sytuacja. Albo więdz podejście ze jedno wspólne repozytorium, albo porozdzielane. Jedno i drugie ma zalety i wady, ale tak to ostro jedziesz po zasobach dyskowych